api/operator_autopilot.go

```diff
@@ -130,19 +130,28 @@ func (d *ReadableDuration) MarshalJSON() ([]byte, error) {
 	return []byte(fmt.Sprintf(`"%s"`, d.Duration().String())), nil
 }
 
-func (d *ReadableDuration) UnmarshalJSON(raw []byte) error {
+func (d *ReadableDuration) UnmarshalJSON(raw []byte) (err error) {
 	if d == nil {
 		return fmt.Errorf("cannot unmarshal to nil pointer")
 	}
 
+	var dur time.Duration
 	str := string(raw)
-	if len(str) < 2 || str[0] != '"' || str[len(str)-1] != '"' {
-		return fmt.Errorf("must be enclosed with quotes: %s", str)
-	}
-	dur, err := time.ParseDuration(str[1 : len(str)-1])
-	if err != nil {
-		return err
+	if len(str) >= 2 && str[0] == '"' && str[len(str)-1] == '"' {
+		// quoted string
+		dur, err = time.ParseDuration(str[1 : len(str)-1])
+		if err != nil {
+			return err
+		}
+	} else {
+		// no quotes, not a string
+		v, err := strconv.ParseFloat(str, 64)
```

Comment:
Ditto: I don't believe handling floats in the api package should be necessary as only the canonical form should be generated by the Consul HTTP API.

Reply:
see comment above re: json encoder emits float64 for a time.Duration by default. LMK what you think.